ethsigner/signer/multiplatform/src/main/java/tech/pegasys/ethsigner/signer/multiplatform/MultiPlatformTransactionSignerProvider.java

```diff
@@ -25,32 +29,62 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class MultiPlatformTransactionSignerProvider implements TransactionSignerProvider {
+public class MultiPlatformTransactionSignerProvider
+    implements TransactionSignerProvider, MultiSignerFactory {
 
   private static final Logger LOG = LogManager.getLogger();
 
   private final SigningMetadataTomlConfigLoader signingMetadataTomlConfigLoader;
+  private final AzureKeyVaultTransactionSignerFactory azureFactory;
 
   MultiPlatformTransactionSignerProvider(
       final SigningMetadataTomlConfigLoader signingMetadataTomlConfigLoader) {
     this.signingMetadataTomlConfigLoader = signingMetadataTomlConfigLoader;
+
+    final AzureKeyVaultAuthenticator azureAuthenticator = new AzureKeyVaultAuthenticator();
+    azureFactory = new AzureKeyVaultTransactionSignerFactory(azureAuthenticator);
   }
 
   @Override
   public Optional<TransactionSigner> getSigner(final String address) {
-    return signingMetadataTomlConfigLoader.loadMetadataForAddress(address).map(this::createSigner);
+    return signingMetadataTomlConfigLoader
+        .loadMetadataForAddress(address)
+        .map(metaDataFile -> metaDataFile.createSigner(this));
   }
 
   @Override
   public Set<String> availableAddresses() {
     return signingMetadataTomlConfigLoader.loadAvailableSigningMetadataTomlConfigs().stream()
-        .map(this::createSigner)
+        .map(metaDataFile -> metaDataFile.createSigner(this))
         .filter(Objects::nonNull)
         .map(TransactionSigner::getAddress)
         .collect(Collectors.toSet());
   }
 
-  private TransactionSigner createSigner(final FileBasedSigningMetadataFile signingMetadataFile) {
+  @Override
+  public TransactionSigner createSigner(final AzureSigningMetadataFile metaDataFile) {
+    final TransactionSigner signer;
+    try {
+      signer = azureFactory.createSigner(metaDataFile.getConfig());
+    } catch (final TransactionSignerInitializationException e) {
+      LOG.error("Failed to construct azure siger from {}", metaDataFile.getBaseFilename());
+      return null;
+    }
+
+    final String signerAddress = signer.getAddress();
+    if (!signerAddress.equals(metaDataFile.getBaseFilename())) {
+      LOG.error(
+          "Azure Signer's Ethereum Address ({}) does not align with metadata filename ({})",
```

Comment:
how would this happen?

Reply:
a badly named file - i.e. a mismatch between filenaming, and the content therein.